buildlib/az-helpers.sh

```diff
@@ -140,6 +140,13 @@ function az_module_unload() {
     module unload "${module}" || true
 }
 
+# Ensure that GPU is present
+check_gpu() {
+    if [ "$(nvidia-smi -L | wc -l)" -eq 0 ]; then
```

Comment:
I'd also add `command -v nvidia-smi` to make sure CUDA toolkit is installed

Reply:
A function 'try_load_cuda_env' runs nvidia-smi already, so this is covered.